src/test/java/net/snowflake/ingest/TestUtils.java

```diff
@@ -317,4 +322,27 @@ public static SimpleIngestManager getManagerUsingBuilderPattern(
         .setUserAgentSuffix(userAgentSuffix)
         .build();
   }
+
+  /**
+   * Given a channel and expected offset, this method waits until the last committed offset is equal
+   * to the passed offset
+   */
+  public static void waitForOffset(SnowflakeStreamingIngestChannel channel, String expectedOffset)
+      throws InterruptedException {
+    int counter = 0;
+    String lastCommittedOffset = null;
+    while (counter < 600) {
+      String currentOffset = channel.getLatestCommittedOffsetToken();
+      if (Objects.equals(expectedOffset, currentOffset)) {
```

Comment:
Can we just not do a regular string comparison here?

Reply:
Fixed